src/sidebar/components/ModerationBanner.js

```diff
@@ -30,14 +30,22 @@ function ModerationBanner({ annotation, api, toastMessenger }) {
   const isHiddenOrFlagged =
     flagCount !== null && (flagCount > 0 || annotation.hidden);
 
+  if (!isHiddenOrFlagged) {
+    return null;
+  }
```

Comment:
Can you talk me through this change? What was the behavior before? The PR description mentions:

> An implicit assumption about annotations already being saved in ModerationBanner was discovered

I have a sense this is related but would like to understand it better.

Reply:
`annotation.id` is undefined if the annotation is not saved. Various parts of this code tried to call API methods like `this.api.annotations.hide({ id: annotation.id})`. For this to work, the annotation must have an ID. Since API method calls now validate that parameter values are strings, numbers, booleans or an array of those, this catches the issue.